src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -238,6 +273,49 @@ static int IsJoystick(const char *path, int fd, char **name_return, SDL_Joystick
     return 1;
 }
 
+static int IsSensor(const char *path, int fd)
+{
+    struct input_id inpid;
+    char *name;
+    char product_string[128];
+
+    if (ioctl(fd, EVIOCGID, &inpid) < 0) {
+        return 0;
+    }
+
+    if (ioctl(fd, EVIOCGNAME(sizeof(product_string)), product_string) < 0) {
```

Comment:
If we're going to apply heuristics to the `EVIOCGNAME` as a source of machine-readable information, then we should have a field for it in the test data in `tests/testevdev.c`. We can leave it as `NULL` for all devices for now - but I can probably populate most or even all of them from the source `evemu-describe` and `steam-runtime-input-monitor` logs.

Reply:
I use SDL_CreateJoystickName() and other parts of SDL use this name to check gamepad support. However if a device has no sensor it should not be added as a sensor even if it matches by name. I'll fix that.